src/lib/output/plugins/HierarchyPlugin.ts

```diff
@@ -0,0 +1,59 @@
+import * as Path from "path";
+import { Component, RendererComponent } from "../components";
+import { RendererEvent } from "../events";
+import { writeFile } from "../../utils";
+import { DefaultTheme } from "../themes/default/DefaultTheme";
+import { gzip } from "zlib";
+import { promisify } from "util";
+import { type Reflection } from "../../models";
+import { UrlMapping } from "../models/UrlMapping";
+
+const gzipP = promisify(gzip);
+
+@Component({ name: "hierarchy" })
+export class HierarchyPlugin extends RendererComponent {
+    override initialize() {
+        this.owner.on(RendererEvent.BEGIN, this.onRendererBegin.bind(this));
+    }
+
+    private onRendererBegin(_event: RendererEvent) {
+        if (!(this.owner.theme instanceof DefaultTheme)) {
+            return;
+        }
+
+        this.owner.preRenderAsyncJobs.push((event) =>
+            this.buildHierarchy(event),
+        );
+    }
+
+    private async buildHierarchy(event: RendererEvent) {
+        const theme = this.owner.theme as DefaultTheme;
+
+        const [_, pageEvent] = event.createPageEvent<Reflection>(
+            new UrlMapping<Reflection>(
+                "assets/hierarchy.js",
+                event.project,
+                () => "",
+            ),
+        );
+
+        const context = theme.getRenderContext(pageEvent);
+
+        const hierarchy = context.getHierarchy();
+
+        const hierarchyJs = Path.join(
```

Comment:
It'd be nice to skip writing this if it is empty, and also skip loading a script to handle it.

Reply:
Smart.